gubernator/view_base.py

```diff
@@ -120,6 +120,18 @@ def gcs_ls(path):
         path += '/'
     return list(gcs.listbucket(path, delimiter='/'))
 
+@memcache_memoize('gs-ls-recursive://', expires=60)
+def gcs_ls_recursive(path):
+    """Enumerate files in a GCS directory recursively. Returns a list of FileStats."""
+    if path[-1] != '/':
+        path += '/'
+
+    files = list(gcs.listbucket(path, delimiter='/'))
```

Comment:
This doesn't have to be recursive. Remove the delimiter and you'll get a listing of *all* files. GCS doesn't have directories, just k/v pairs!

Reply:
I had tried that but I thought I got the entire bucket, not just what was under the path. I'll change to using that functionality, though, as it's probably faster.